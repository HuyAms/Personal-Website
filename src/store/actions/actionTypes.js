export const FETCH_PROJECTS_START = "FETCH_PROJECTS_START";
export const FETCH_PROJECTS_SUCCESS = "FETCH_PROJECTS_SUCCESS";
export const FETCH_PROJECT_FAIL = "FETCH_PROJECT_FAIL";
export const SET_PROJECT_FILTER = "SET_PROJECT_FILTER";
export const SET_PROJECT_SERACH = "SET_PROJECT_SEARCH";

export const FETCH_INFO_START = "FETCH_INFO_START";
export const FETCH_INFO_SUCCESS = "FETCH_INFO_SUCCESS";
export const FETCH_INFO_FAIL = "FETCH_INFO_FAIL";

export const FETCH_POSITIVE_START = "FETCH_POSITIVE_START";
export const FETCH_POSITIVE_SUCCESS = "FETCH_POSITIVE_SUCCESS";
export const FETCH_POSITIVE_FAIL = "FETCH_POSITIVE_FAIL";

export const POST_POSITIVE_START = "POST_POSITIVE_START";
export const POST_POSITIVE_SUCCESS = "POST_POSITIVE_SUCCESS";
export const POST_POSITIVE_FAIL = "POST_POSITIVE_FAIL";
export const OPEN_POSITIVE_FORM = "OPEN_POSITIVE_FORM";
export const CLOSE_POSITIVE_FORM = "CLOSE_POSITIVE_FORM";