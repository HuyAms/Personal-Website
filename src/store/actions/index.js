export {
  fetchProject
} from './project';

export {
  fetchInfo
} from './info';