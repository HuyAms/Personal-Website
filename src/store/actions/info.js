import * as actionTypes from './actionTypes';

export const fetchInfoSuccess = (myInfo) => {
  return {
    type: actionTypes.FETCH_INFO_SUCCESS,
    myInfo
  }
}

export const fetchInfoFail = (error) => {
  return {
    type: actionTypes.FETCH_INFO_FAIL,
    error
  }
}

export const fetchInfoStart = () => {
  return {
    type: actionTypes.FETCH_INFO_START,
  }
}

export const fetchExpSuccess = (workExps) => {
  return {
    type: actionTypes.FETCH_EXP_SUCCESS,
    workExps
  }
}

export const fetchExpFail = (error) => {
  return {
    type: actionTypes.FETCH_EXP_FAIL,
    error
  }
}

export const fetchExpStart = () => {
  return {
    type: actionTypes.FETCH_EXP_START,
  }
}
