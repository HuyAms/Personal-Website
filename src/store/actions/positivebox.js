import * as actionTypes from './actionTypes';
import axios from '../../axious';

export const fetchPositiveSuccess = (positives) => {
  return {
    type: actionTypes.FETCH_POSITIVE_SUCCESS,
    positives: positives,
  };
};

export const fetchPositiveFail = (error) => {
  return {
    type: actionTypes.FETCH_POSITIVE_FAIL,
    error: error,
  };
};

export const fetchPositiveStart = () => {
  return {
    type: actionTypes.FETCH_POSITIVE_START,
  };
};

export const postPositiveSuccess = (positiveData, id) => {
  return {
    type: actionTypes.POST_POSITIVE_SUCCESS,
    positiveData: positiveData,
    id: id,
  };
};

export const postPositiveFail = (error) => {
  return {
    type: actionTypes.POST_POSITIVE_FAIL,
    error: error,
  };
};

export const postPositiveStart = () => {
  return {
    type: actionTypes.POST_POSITIVE_START,
  };
};

export const openPositiveForm = () => {
  return {
    type: actionTypes.OPEN_POSITIVE_FORM
  }
};

export const closePositiveForm = () => {
  return {
    type: actionTypes.CLOSE_POSITIVE_FORM
  }
};

export const fetchPositive = () => {
  return dispatch => {
    dispatch(fetchPositiveStart());
    axios.get('/positive.json').then(res => {
      const fetchedPositive = [];
      for (let key in res.data) {
        fetchedPositive.push({
          ...res.data[key],
          id: key,
        });
      }
      dispatch(fetchPositiveSuccess(fetchedPositive));
    }).catch(error => {
      dispatch(fetchPositiveFail(error));
    });
  };
};

export const postPositives = (positiveData) => {
  return dispatch => {
    dispatch(postPositiveStart());
    axios.post('/positive.json', positiveData).then(res => {
      dispatch(postPositiveSuccess(positiveData, res.data.name));
    }).catch(error => {
      dispatch(postPositiveFail(error));
    });
  };
};