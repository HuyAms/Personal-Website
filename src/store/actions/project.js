import * as actionTypes from './actionTypes';
import axios from '../../axious';

export const fetchProjectsSuccess = (projects) => {
  return {
    type: actionTypes.FETCH_PROJECTS_SUCCESS,
    projects: projects
  }
}

export const fetchProjectsFail = (error) => {
  return {
    type: actionTypes.FETCH_PROJECT_FAIL,
    error: error
  }
}

export const fetchProjectsStart = () => {
  return {
    type: actionTypes.FETCH_PROJECTS_START
  }
}

export const setProjectFilter = (filter) => {
  return {
    type: actionTypes.SET_PROJECT_FILTER,
    filter: filter
  }
}

export const fetchProject = () => {
  return dispatch => {
    dispatch(fetchProjectsStart());
    axios.get('/projects.json')
    .then(res => {
      const fetchedProject = [];
      for (let key in res.data) {
        fetchedProject.push({
          ...res.data[key],
          id: key,
        });
      }
      dispatch(fetchProjectsSuccess(fetchedProject))
    })
    .catch(error => {
        dispatch(fetchProjectsFail(error))
    })
  }
}
