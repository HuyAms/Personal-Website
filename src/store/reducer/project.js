import * as actionTypes from '../actions/actionTypes';
import updateObject from '../utility';

const initialState = {
  projects: [],
  loading: false,
  error: null,
  activeFilterIndex: 0,
  search: '',
  h2ON32: false
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case actionTypes.FETCH_PROJECTS_START:
      return fetchProjectStart(state, action);
    case actionTypes.FETCH_PROJECT_FAIL:
      return fetchProjectFail(state, action);
    case actionTypes.FETCH_PROJECTS_SUCCESS:
      return fetchProjectSuccess(state, action);
    case actionTypes.SET_PROJECT_FILTER:
      return setProjectFilter(state, action);
    case actionTypes.SET_PROJECT_SERACH:
      return setProjectSearch(state, action);
    default:
      return state;
  }
};

const setProjectSearch = (state, action) => {
  if (action.search.toLowerCase() === 'nhung') {
    return updateObject(state, {
      search: action.search,
      h2ON32: true
    })
  }
  return updateObject(state, {
    search: action.search,
    h2ON32: false
  })
}

const setProjectFilter = (state, action) => {
  return updateObject(state, {
    activeFilterIndex: action.filter
  })
}

const fetchProjectStart = (state, action) => {
  return updateObject(state, {
    error: null,
    loading: true,
  });
};

const fetchProjectFail = (state, action) => {
  return updateObject(state, {
    error: action.error,
    loading: false,
  });
};

const fetchProjectSuccess = (state, action) => {
  return updateObject(state, {
    projects: action.projects,
    loading: false,
  });
};

export default reducer;