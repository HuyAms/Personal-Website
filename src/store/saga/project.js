import {put, call} from 'redux-saga/effects';
import * as actions from '../actions/project';
import axios from "../../axious";

export function* fetchProjectSaga(action) {
  try {

    const response = yield call(axios.get, '/projects.json');

    const fetchedProject = [];
    for (let key in response.data) {
      fetchedProject.push({
        ...response.data[key],
        id: key,
      });
    }

    yield put(actions.fetchProjectsSuccess(fetchedProject));

  } catch(error) {

    yield put(actions.fetchProjectsFail(error));
  }
}
