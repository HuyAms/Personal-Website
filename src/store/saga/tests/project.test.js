/**
 * Tests for Project sagas
 */

import {put} from 'redux-saga/effects';
import {fetchProjectSaga} from '../project';
import * as actions from '../../actions/project';

describe('fetchProjectSaga', () => {
  let fetchProjectGenerator;

  beforeEach(() => {
    fetchProjectGenerator = fetchProjectSaga();

    const callDescriptor = fetchProjectGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the fetchProjectsSuccess action if it requests the data successfully', () => {
    const response = {
      data: []
    }

    const putDescriptor = fetchProjectGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(actions.fetchProjectsSuccess(response.data)));
  })

  it('should dispatch the fetchProjectsFail action if the response errors', () => {
    const error = new Error('Some error');
    const putDescriptor = fetchProjectGenerator.throw(error).value;
    expect(putDescriptor).toEqual(put(actions.fetchProjectsFail(error)));
  })
})