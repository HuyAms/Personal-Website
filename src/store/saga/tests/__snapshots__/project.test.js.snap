// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchProjectSaga should dispatch the fetchProjectsFail action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/projects.json",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`fetchProjectSaga should dispatch the fetchProjectsSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/projects.json",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
